src/GithubPlugin/Providers/SettingsUIController.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using GitHubPlugin.Helpers;
+using Microsoft.Windows.ApplicationModel.Resources;
+using Microsoft.Windows.DevHome.SDK;
+using Newtonsoft.Json.Linq;
+using Windows.Foundation;
+
+namespace GitHubPlugin.Providers;
+internal class SettingsUIController : IExtensionAdaptiveCardSession
+{
+    private IExtensionAdaptiveCard? _settingsUI;
+    private static readonly SettingsUITemplate _settingsUITemplate = new ();
+
+    public SettingsUIController()
+    {
+    }
+
+    public void Dispose()
+    {
+        Log.Logger()?.ReportDebug($"Dispose");
+        _settingsUI?.Update(null, null, null);
+    }
+
+    public ProviderOperationResult Initialize(IExtensionAdaptiveCard extensionUI)
+    {
+        Log.Logger()?.ReportDebug($"Initialize");
+        _settingsUI = extensionUI;
+        return _settingsUI.Update(_settingsUITemplate.GetSettingsUITemplate(), null, "SettingsPage");
+    }
+
+    public IAsyncOperation<ProviderOperationResult> OnAction(string action, string inputs)
+    {
+        return Task.Run(async () =>
+        {
+            ProviderOperationResult operationResult;
+            Log.Logger()?.ReportInfo($"OnAction() called with state:{_settingsUI?.State}");
+            Log.Logger()?.ReportDebug($"action: {action}");
+
+            switch (_settingsUI?.State)
+            {
+                case "SettingsPage":
+                    {
+                        Log.Logger()?.ReportDebug($"inputs: {inputs}");
+
+                        var currentNotificationsEnabled = LocalSettings.ReadSettingAsync<string>("NotificationsEnabled").Result ?? "true";
+                        await LocalSettings.SaveSettingAsync("NotificationsEnabled", currentNotificationsEnabled == "true" ? "false" : "true");

```

Comment:
Nit: make these strings variables that can be reused.

Reply:
Fixed.